### PR TITLE
[Incubator-kie-issues#1356] DMN Elements with null IDs are not correctly managed

### LINKED ISSUE #1356: DMN Elements with null IDs are not correctly managed
DMN spec states that the `ID` attribute of any `DMNElements` is Optional

Screenshot 2024-06-25 at 10 16 29

This is currently not correctly managed in our code base.

The DMNElements are managed in the `DMNModelImpl` in `Maps`

Screenshot 2024-06-25 at 10 19 53

Those maps are managed with ID as key.
In case of multiple DMNElements of the same kind (eg. Decision Nodes) with no ID, all those nodes will be added in that Map with `null` key, resulting in a collision.

We should manage this case using a new key (namespace + name), but I guess we should keep the search by `ID` in place, to preserve the API backword-compatibility

### PR DESCRIPTION
**Pull request by gitgabrio**
Fixes https://github.com/apache/incubator-kie-issues/issues/1356

@bncriju

<details>
<summary>
How to replicate CI configuration locally?
</summary>

Build Chain tool does "simple" maven build(s), the builds are just Maven commands, but because the repositories relates and depends on each other and any change in API or class method could affect several of those repositories there is a need to use [build-chain tool](https://github.com/kiegroup/github-action-build-chain) to handle cross repository builds and be sure that we always use latest version of the code for each repository.
 
[build-chain tool](https://github.com/kiegroup/github-action-build-chain) is a build tool which can be used on command line locally or in Github Actions workflow(s), in case you need to change multiple repositories and send multiple dependent pull requests related with a change you can easily reproduce the same build by executing it on Github hosted environment or locally in your development environment. See [local execution](https://github.com/kiegroup/github-action-build-chain#local-execution) details to get more information about it.
</details>

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

- for <b>pull request and downstream checks</b>  
  - Push a new commit to the PR. An empty commit would be enough.

- for a <b>full downstream build</b>
  - for <b>github actions</b> job: add the label `run_fdb`

- for <b>Jenkins PR check only</b>
  - If you are an ASF committer for KIE podling, login to Jenkins (https://ci-builds.apache.org/job/KIE/job/drools/), go to the specific PR job, and click on `Build Now` button.
</details>

### CODE REVIEW

**Review by yesamer — CHANGES_REQUESTED** [review]
With this PR you solve 24 TCK Tests.
Unfortunately, it introduced a regression. 
This is the case failing after applying the changes
https://github.com/dmn-tck/tck/tree/master/TestCases/compliance-level-3/0089-nested-inputdata-imports
It seems related to Imported DMN.



`09:39:26.768 [main] ERROR org.kie.dmn.core.util.MsgUtil.logMessage:83 - Required import not found: {http://www.trisotech.com/definitions/_ae5b3c17-1ac3-4e1d-b4f9-2cf861aec6d9}Say hello 1ID1D for node 'Model C'; missing DMN Import name alias.
09:39:26.769 [main] ERROR org.kie.dmn.core.util.MsgUtil.logMessage:83 - Required dependency 'Person name' not found on node 'Greet the Person'
09:39:26.769 [main] INFO  o.o.d.t.runner.drools.DroolsTCKTest.executeTest:271 - Messages: 
-----
Message [id=0, level=ERROR, path=/Users/yamer/Projects/tck/runners/dmn-tck-runner-drools/../../TestCases/compliance-level-3/0089-nested-inputdata-imports/0089-nested-inputdata-imports.dmn, line=19, column=-1
   text=DMN: Required import not found: {http://www.trisotech.com/definitions/_ae5b3c17-1ac3-4e1d-b4f9-2cf861aec6d9}Say hello 1ID1D for node 'Model C'; missing DMN Import name alias. (DMN id: _10435dcd-8774-4575-a338-49dd554a0928, The referenced import was not found) ]
Message [id=0, level=ERROR, path=/Users/yamer/Projects/tck/runners/dmn-tck-runner-drools/../../TestCases/compliance-level-3/0089-nested-inputdata-imports/0089-nested-inputdata-imports.dmn, line=21, column=-1
   text=DMN: Required dependency 'Person name' not found on node 'Greet the Person' (DMN id: _f7fdaec4-d669-4797-b3b4-12b860de2eb5, The referenced node was not found) ]
-----

09:39:26.770 [main] INFO  o.o.d.t.runner.drools.DroolsTCKTest.executeTest:310 - Result context: {
    Model B: null
    Model B2: null
    modelA: {
    }
}`

In addition, there are unneeded changes in the file's headers

File: kie-dmn/kie-dmn-core/src/test/java/org/kie/dmn/core/impl/TupleIdentifierTest.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+import static org.junit.jupiter.api.Assertions.assertNull;
+
+class TupleIdentifierTest {
+
+    @Test
+    void createTupleIdentifierById() {
+        String name = "name";
+        String wrongName = "wrong-name";
+        String retrieved = TupleIdentifier.generateIdFromName(name);
+        assertEquals(retrieved, TupleIdentifier.generateIdFromName(name));
```
Comment (pibizza) [inline kie-dmn/kie-dmn-core/src/test/java/org/kie/dmn/core/impl/TupleIdentifierTest.java]: If possible use assertj to perform assertions. We have made in the past a rather large effort to use only assertj in drools to do assertions.

**Comment by gitgabrio** [comment]
> With this PR you solve 24 TCK Tests. Unfortunately, it introduced a regression. This is the case failing after applying the changes https://github.com/dmn-tck/tck/tree/master/TestCases/compliance-level-3/0089-nested-inputdata-imports It seems related to Imported DMN.
> ## `09:39:26.768 [main] ERROR org.kie.dmn.core.util.MsgUtil.logMessage:83 - Required import not found: {[http://www.trisotech.com/definitions/_ae5b3c17-1ac3-4e1d-b4f9-2cf861aec6d9}Say](http://www.trisotech.com/definitions/_ae5b3c17-1ac3-4e1d-b4f9-2cf861aec6d9%7DSay) hello 1ID1D for node 'Model C'; missing DMN Import name alias.
> 09:39:26.769 [main] ERROR org.kie.dmn.core.util.MsgUtil.logMessage:83 - Required dependency 'Person name' not found on node 'Greet the Person'
> 09:39:26.769 [main] INFO o.o.d.t.runner.drools.DroolsTCKTest.executeTest:271 - Messages:
> ## Message [id=0, level=ERROR, path=/Users/yamer/Projects/tck/runners/dmn-tck-runner-drools/../../TestCases/compliance-level-3/0089-nested-inputdata-imports/0089-nested-inputdata-imports.dmn, line=19, column=-1
> text=DMN: Required import not found: {http://www.trisotech.com/definitions/_ae5b3c17-1ac3-4e1d-b4f9-2cf861aec6d9}Say hello 1ID1D for node 'Model C'; missing DMN Import name alias. (DMN id: _10435dcd-8774-4575-a338-49dd554a0928, The referenced import was not found) ]
> Message [id=0, level=ERROR, path=/Users/yamer/Projects/tck/runners/dmn-tck-runner-drools/../../TestCases/compliance-level-3/0089-nested-inputdata-imports/0089-nested-inputdata-imports.dmn, line=21, column=-1
> text=DMN: Required dependency 'Person name' not found on node 'Greet the Person' (DMN id: _f7fdaec4-d669-4797-b3b4-12b860de2eb5, The referenced node was not found) ]
> 
> 09:39:26.770 [main] INFO o.o.d.t.runner.drools.DroolsTCKTest.executeTest:310 - Result context: { Model B: null Model B2: null modelA: { } }`
> 
> In addition, there are unneeded changes in the file's headers

Do you know hat model this `[import](https://github.com/dmn-tck/tck/blob/9d2b7919896f3eaf5f9658097fe0a540e205c86f/TestCases/compliance-level-3/0089-nested-inputdata-imports/Model_B.dmn#L19)` is referring to ? (I found it also in other dmns in that test)

**Comment by yesamer** [comment]
@gitgabrio 24 TCK Tests are now green. Well done!

**Review by baldimir — APPROVED** [review]

**Review by yesamer — APPROVED** [review]
LGTM

**Pull request merged**